Tests/Serializer/BaseSerializerHandlerTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Tests\Serializer;
+
+use JMS\Serializer\GraphNavigator;
+use Sonata\CoreBundle\Tests\Fixtures\Bundle\Serializer\FooSerializer;
+
+/**
+ * @author Ahmet Akbana <ahmetakbana@gmail.com>
+ */
+final class BaseSerializerHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetSubscribingMethods()
+    {
+        $manager = $this->getMock('Sonata\CoreBundle\Model\ManagerInterface');
+
+        $serializer = new FooSerializer($manager);
+
+        $expectedMethods = array(
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'json',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'json',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+        );
+
+        $methods = $serializer->getSubscribingMethods();
+
+        $this->assertSame($methods, $expectedMethods);
```

Comment:
Doesn't this test feel weird? If yes, it is probably because the formats are hardcoded in the class. Improve that if you feel up to it (by introducing a `formats` property that you might change from your tests), and it the end, the tests should look like this : 
- check that there are twice as many methods as the number of formats
- check that half for them are about deserialization…